server/src/test/java/org/opensearch/plugins/PluginsServiceTests.java

```diff
@@ -717,6 +718,19 @@ public void testIncompatibleOpenSearchVersion() throws Exception {
         assertThat(e.getMessage(), containsString("was built for OpenSearch version 6.0.0"));
     }
 
+    public void testPluginDifferingWithCoreForPatchVersion() throws Exception {
+        // Get a version for plugin to use that is just before the core's version.
+        // We will use the plugin version for test only if it differs in patch version
+        Version pluginVersion = VersionUtils.getPreviousVersion(Version.CURRENT);
+        if (pluginVersion.major != Version.CURRENT.major || pluginVersion.minor != Version.CURRENT.minor) {
```

Comment:
This doesn't look like a predictable test. Write tests that have predictable versions of the hypothetical plugin and cover all scenarios.

Reply:
Added predictable tests.